toolkit/primitives/authority-selection-inherents/src/filter_invalid_candidates.rs

```diff
@@ -31,18 +31,29 @@ pub struct CandidateWithStake<TAccountId, TAccountKeys> {
 	pub account_keys: TAccountKeys,
 }
 
-#[derive(Clone, Debug, Encode, Decode, PartialEq)]
+#[derive(Clone, Debug, Encode, Decode, PartialEq, Eq)]
 pub struct PermissionedCandidate<TAccountId, TAccountKeys> {
 	pub account_id: TAccountId,
 	pub account_keys: TAccountKeys,
 }
 
-#[derive(Clone, Debug, Encode, Decode, PartialEq)]
+#[derive(Clone, Debug, Encode, Decode, PartialEq, Eq)]
 pub enum Candidate<TAccountId, TAccountKeys> {
 	Permissioned(PermissionedCandidate<TAccountId, TAccountKeys>),
 	Registered(CandidateWithStake<TAccountId, TAccountKeys>),
 }
 
+impl<TAccountId: Ord + Eq, TAccountKeys: Eq> PartialOrd for Candidate<TAccountId, TAccountKeys> {
+	fn partial_cmp(&self, other: &Self) -> Option<core::cmp::Ordering> {
+		Some(self.cmp(other))
+	}
+}
+impl<TAccountId: Ord + Eq, TAccountKeys: Eq> Ord for Candidate<TAccountId, TAccountKeys> {
+	fn cmp(&self, other: &Self) -> core::cmp::Ordering {
+		self.account_id().cmp(other.account_id())
+	}
+}
```

Comment:
there is no need to restrict Eq, PartialOrd/Ord implies that
```suggestion
impl<TAccountId: Ord, TAccountKeys: Eq> PartialOrd for Candidate<TAccountId, TAccountKeys> {
	fn partial_cmp(&self, other: &Self) -> Option<core::cmp::Ordering> {
		Some(self.cmp(other))
	}
}
impl<TAccountId: Ord, TAccountKeys: Eq> Ord for Candidate<TAccountId, TAccountKeys> {
	fn cmp(&self, other: &Self) -> core::cmp::Ordering {
		self.account_id().cmp(other.account_id())
	}
}
```